### DISCUSSION TITLE
Modularity & Ability to create and use model at runtime.

### DISCUSSION THREAD

**Discussion by reggi** [prisma/prisma#15432] [Q&A] [answered]
I love prisma, I think it does everything very well. I do think that the prisma file itself and the type definitions are also great.  However, I'm interested in the ability to have modules / packages / plugins that create their own tables / models programatically. As far as I know the only way currently to interact with database tables is by defining them before runtime generating the typescript code necessary to call the database models, and using it. I would like the ability to create and use it more dynamically without the need for a build step. One way to think of this is having a npm package with a model / table definition in it. How could we do that with prisma? 

Sort of related to https://github.com/prisma/prisma/issues/2377

**Comment by austincrim** [comment, marked as answer]
Hey @reggi 👋,

Your request is not uncommon, and I think the feature request you linked will be the long-term solution. Other ideas are things like [code-first schema definition](https://github.com/prisma/prisma/issues/227).

We always love to hear new use cases and feedback, so please leave your thoughts on either issue. This also helps our engineers prioritize features for the future.

**Comment by ZhaoKunLong** [comment]
Me too! I also have this requirement, My case is I got a `rfidHistory` to store the sensor data, each 10s will have 20 records. One day in my case will have 4 million data,  one month will have 120 million data,   I have split them into each day's table like `rfidHistory_{day}`.  I love to use the Prisma model to do the data query and manage. But I can not find a way that use Prisma to create a new table and manage it. 
Most articles say I have to do it by using SQL. It's not friendly to me. 
Do you have any ideas or solutions for this? Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
